projects/fusion-ui/components/chip-filter/common/base/chip-filter.base.component.ts

```diff
@@ -0,0 +1,102 @@
+import {Directive, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output, Renderer2, ViewChild} from '@angular/core';
+import {fromEvent, Subject} from 'rxjs';
+import {ChipFilterComponentConfigurations, ChipType, ChipTypeToClass} from './chip-filter-component-configurations';
+import {takeUntil} from 'rxjs/operators';
+
+@Directive()
+export abstract class ChipFilterBaseComponent implements OnInit, OnDestroy {
+    id: number | string;
+    width: number;
+
+    private _selected: boolean;
+    private _disabled: boolean;
+    private _close: boolean;
+    onDestroy$ = new Subject<void>();
+
+    @Input() set configuration(value: ChipFilterComponentConfigurations) {
+        if (!!value) {
+            this.id = value.id;
+            this.close = value.close;
+            this.disabled = value.disabled;
+            this.selected = value.selected;
+            this.type = value.type;
+        }
+    }
+    @Input() suppressClickOnRemove = false;
+
+    @Input() set close(close: boolean) {
+        this._close = close;
+        this.changeHostClass('with-closed-icon', close);
+    }
+
+    get close(): boolean {
+        return this._close;
+    }
+
+    @Input() set disabled(disabled: boolean) {
+        this._disabled = disabled;
+        this.changeHostClass('fu-disabled', disabled);
+    }
+
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    @Input()
```

Comment:
why do you have also @input for specific configuration and also for @input configuration